From fd05209f21bcdb259cfd4b727d2582fc8e0d4b08 Mon Sep 17 00:00:00 2001
From: Gabriel Fernandez <gabriel.fernandez@foss.st.com>
Date: Fri, 20 Sep 2024 18:01:34 +0200
Subject: [PATCH 1249/1695] dts: stm32: add SCMI HSI/HSE/MSI Kernel clocks for
 stm32mp25x

Adds CK_SCMI_HSI_KER_CK, CK_SCMI_HSE_KER_CK and CK_SCMI_MSI_KER_CK clocks

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 65022f83f5e57fa91aa52a9ef7d74e037ae5eca0 dts: stm32: add SCMI HSI/HSE/MSI Kernel clocks for stm32mp25x]
Signed-off-by: Gabriel Fernandez <gabriel.fernandez@foss.st.com>
Change-Id: I49b2484649aeec9692d274695a00a3a564d0cb26
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/406046
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
Domain-Review: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
---
 core/arch/arm/dts/stm32mp251.dtsi | 42 +++++++++++++++++++++++++++++++
 1 file changed, 42 insertions(+)

diff --git a/core/arch/arm/dts/stm32mp251.dtsi b/core/arch/arm/dts/stm32mp251.dtsi
index 8c0eaef57..90e2f84a5 100644
--- a/core/arch/arm/dts/stm32mp251.dtsi
+++ b/core/arch/arm/dts/stm32mp251.dtsi
@@ -1045,6 +1045,27 @@
 						clocks = <&rcc CK_KER_ETR>;
 						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
 					};
+
+					clock@7d {
+						reg = <CK_SCMI_HSI_KER_CK>;
+						domain-name = "hsi_ker_ck";
+						clocks = <&rcc HSI_KER_CK>;
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@7e {
+						reg = <CK_SCMI_HSE_KER_CK>;
+						domain-name = "hse_ker_ck";
+						clocks = <&rcc HSE_KER_CK>;
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@7f {
+						reg = <CK_SCMI_MSI_KER_CK>;
+						domain-name = "msi_ker_ck";
+						clocks = <&rcc MSI_KER_CK>;
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
 				};
 			};
 
@@ -2037,6 +2058,27 @@
 						clocks = <&rcc CK_KER_ETR>;
 						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
 					};
+
+					clock@7d {
+						reg = <CK_SCMI_HSI_KER_CK>;
+						domain-name = "hsi_ker_ck";
+						clocks = <&rcc HSI_KER_CK>;
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@7e {
+						reg = <CK_SCMI_HSE_KER_CK>;
+						domain-name = "hse_ker_ck";
+						clocks = <&rcc HSE_KER_CK>;
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@7f {
+						reg = <CK_SCMI_MSI_KER_CK>;
+						domain-name = "msi_ker_ck";
+						clocks = <&rcc MSI_KER_CK>;
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
 				};
 			};
 
-- 
2.47.2

